db-engine/src/main/java/com/splicemachine/db/iapi/types/SQLBit.java

```diff
@@ -229,6 +230,30 @@ else if (sourceWidth > desiredWidth)
 		}
 	}
 
+	@Override
+	public final String toString() {
+		if (dataValue == null)
+		{
+			if ((stream == null) && (_blobValue == null) )
+			{
+				return "NULL";
+			}
+			else
+			{
+				if (SanityManager.DEBUG)
+					SanityManager.THROWASSERT(
+							"value is null, stream or blob is not null");
+				return "";
+			}
+		}
+		else
+		{
+			String hex = com.splicemachine.db.iapi.util.StringUtil.toHexString(dataValue, 0, dataValue.length);
+			return String.format("X'%s'",hex);
```

Comment:
It seems that the only difference between the toString() function in SQLBit and that in SQLBinary is the call to String.format(), why don't we move this implementation up to SQLBinary instead? It seems to make sense to apply String.format() there too.

Reply:
This impacts all SQLBinary's subclasses, include SQLBlob. X'whatever' works for char/varchar/ long char bit data, not for Blob

Reply:
Ok, thanks for the explanation!